### PR TITLE
Side panel

### PR DESCRIPTION
**Pull request by kanastasakis26**
Side panel
Cytoscape
React

### CODE REVIEW

File: client/ts/data.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        .then((response) => {
+          return response.data;
+        })
+        .then((data) => {
+            store.dispatch(saveFilterOptions(getNodeProperties(data)));
+            store.dispatch(saveGraph(convertToCYFormat(data)));
+        })
+}
+
+function getNodeProperties(graphDictionary:any) {
```
Comment (xyz8983) [inline client/ts/data.ts:15]: do you mind adding some docstring?

File: client/ts/index.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+)
+
+
+// For file uploads
+let fileSelectComponent = document.getElementById("upload-box");
+if(fileSelectComponent) {
+    fileSelectComponent.addEventListener('change', (event) => {
+            let fileInput: HTMLInputElement = event.target as HTMLInputElement;
+            let filesList: FileList = fileInput.files;
+            if (filesList.length > 0) {
```
Comment (xyz8983) [inline client/ts/index.tsx:49]: if we change it to upload a folder, will the fileList > 0 still holde?

File: client/ts/panels/sidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import Form from 'react-bootstrap/Form'
+
+import {saveFilterSelections} from '../store'
+
+export interface SidebarProps {
+    filterOptions?:any,
+    saveFilterSelections:any
+}
+
+class Sidebar extends React.Component<SidebarProps> {
```
Comment (xyz8983) [inline client/ts/panels/sidebar.tsx:13]: would you mind add some docsring?

File: client/ts/panels/sidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                    {nodeNames.map((nodeName, i:number) => {
+                        return <option key={`_${i}`}>{nodeName}</option>
+                    })
+                    }
+                    </Form.Control>
+                </Form.Group>
+                {/*<Form.Group>*/}
+                {/*    <Form.Label>Feature type</Form.Label>*/}
+                {/*    <div id={"node-type-checkboxes"}>*/}
+                {/*        {nodeTypeOption.map((type, i) => (*/}
```
Comment (xyz8983) [inline client/ts/panels/sidebar.tsx:73]: is this for displaying checkbox?

Comment (kanastasakis26) [reply client/ts/panels/sidebar.tsx:73]: These for other filter options. I decided to leave them out for now.

File: client/ts/index.tsx (hunk 2 above)
Comment (kanastasakis26) [reply client/ts/index.tsx:49]: Not sure

**Pull request merged**